library/std/src/sys/unix/net.rs

```diff
@@ -376,6 +384,21 @@ impl Socket {
         Ok(())
     }
 
+    pub fn set_linger(&self, linger: Option<Duration>) -> io::Result<()> {
+        let linger = libc::linger {
+            l_onoff: linger.is_some() as libc::c_int,
+            l_linger: linger.map(|dur| dur.as_secs() as libc::c_int).unwrap_or_default(),
```

Comment:
`Duration` implements `Default`:

```suggestion
            l_linger: linger.unwrap_or_default().as_secs() as libc::c_int,
```